app/src/main/java/com/github/polypoly/app/ui/game/HeadsUpDisplay.kt

```diff
@@ -55,14 +64,23 @@ fun HudLocation(location: String) {
             .fillMaxWidth()
             .padding(Padding.medium)
     ) {
-        Text(
-            text = location,
-            modifier = Modifier
-                .align(Alignment.TopCenter).testTag("location_text")
-                .offset(y = 30.dp)
-            ,
-            fontSize = MaterialTheme.typography.h6.fontSize
-        )
+        if (location.isNotEmpty())
+            Text(
+                text = location,
+                modifier = Modifier
+                    .align(Alignment.TopCenter)
+                    .testTag("location_text")
+                    .offset(y = 10.dp)
+                    .background(MaterialTheme.colors.background, shape = Shapes.medium)
+                    .border(
+                        1.dp,
+                        changeOpacity(MaterialTheme.colors.onBackground, 0.5f),
+                        shape = Shapes.medium
+                    )
+                    .padding(Padding.medium),
+                fontSize = MaterialTheme.typography.h4.fontSize,
+                fontWeight = MaterialTheme.typography.h4.fontWeight
```

Comment:
You can use `style = MaterialTheme.typography.h4` instead of the two line above

Reply:
fixed